modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v3_7_1/test/DDMStructureCTUpgradeProcessTest.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.dynamic.data.mapping.upgrade.v3_7_1.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.dynamic.data.lists.model.DDLRecordSet;
+import com.liferay.dynamic.data.mapping.io.DDMFormSerializer;
+import com.liferay.dynamic.data.mapping.model.DDMForm;
+import com.liferay.dynamic.data.mapping.model.DDMStructure;
+import com.liferay.dynamic.data.mapping.service.DDMStructureLocalService;
+import com.liferay.dynamic.data.mapping.test.util.DDMFormTestUtil;
+import com.liferay.dynamic.data.mapping.test.util.DDMStructureTestUtil;
+import com.liferay.dynamic.data.mapping.upgrade.v3_7_1.test.util.DDMStructureCTUpgradeProcessTestUtil;
+import com.liferay.dynamic.data.mapping.util.DDMFormSerializeUtil;
+import com.liferay.portal.kernel.model.change.tracking.CTModel;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.test.rule.Inject;
+
+import org.junit.runner.RunWith;
+
+/**
+ * @author Igor Costa
+ */
+@RunWith(Arquillian.class)
+public class DDMStructureCTUpgradeProcessTest
+	extends com.liferay.dynamic.data.mapping.upgrade.v5_1_1.test.
+				DDMStructureCTUpgradeProcessTest {
+
+	@Override
+	protected CTModel<?> addCTModel() throws Exception {
+		DDMStructure ddmStructure = DDMStructureTestUtil.addStructure(
+			getGroup().getGroupId(), DDLRecordSet.class.getName(),
+			DDMFormTestUtil.createDDMForm(RandomTestUtil.randomString()));
+
+		ddmStructure.setDefinition(
+			DDMFormSerializeUtil.serialize(
+				getDDMForm(), _jsonDDMFormSerializer));
+
+		return _ddmStructureLocalService.updateDDMStructure(ddmStructure);
+	}
```

Comment:
I understand you're having overrode this method because this example uses DDLRecordSet instead of DDMFormInstance. But there is a way to circumvent this situation, so we won't need to override this.

Reply:
In the class v5_1_1/test/DDMStructureCTUpgradeProcessTest.java:

- create a method named getClassName() and it will return DDMFormInstance.class.getName().
- rename the previous method getClassName() to getUpgradeStepClassName()
- use the method getClassName() in the  addCTModel()

Reply:
In the class DDMStructureCTUpgradeProcessTestUtil ( all the existing and new util):

- rename the previous method getClassName() to getUpgradeStepClassName()

Reply:
In the class v3_7_1/test/DDMStructureCTUpgradeProcessTest.java:

- Override getClassName() to return DDLRecordSet.class.getName().